gateway/1.5/modules/ROOT/pages/policies-pdk-create-schema-definition.adoc

```diff
@@ -1,5 +1,362 @@
++
+Defining a string as the `uri` or `ip-range` provides additional validations, but defining a string as `dataweave` notifies Flex Gateway that the input must be transformed. For more information about supported DataWeave transformations, see xref:policies-flex-dataweave-support.adoc[].
++
+The following yaml snippet demonstrates all string formats:
++
+[source,yaml]
+----
+properties:
+    loginUrl:
+        type: string
+        format: uri
+        default: "https://anypoint.mulesoft.com/login"
+    blockedIps:
+        type: string
+        format: ip-range
+        default: "192.168.3.1/30"
+    token:
+	type: string
+        format: dataweave
+        default: "#[splitBy(attributes.headers['Authorization'], ' ')[1]]"
+----
++
+Number::
++
+The `number` type specifies data as a floating point value.
++
+For example:
++
+[source,yaml]
+----
+properties:
+    squareRoot:
+        type: number
+        default: 1.414213562
+    seconds:
+        type: number
+        default: 10
+----
++
+Integer::
++
+The `integer` type specifies data as an integer value.
++
+For example:
++
+[source,yaml]
+----
+properties:
+    maxHeaders:
+        type: integer
+        default: 10
+----
++
+Boolean::
++
+The `boolean` type specifies data as a boolean value. Supported values are `true` or `false`.
+For example:
++
+[source,yaml]
+----
+properties:
+    skipValidation:
+        type: boolean
+        default: false
+    requireCredentials:
+        type: boolean
+        default: true
+----
++
+Array::
++
+The `array` type specifies the data as a list of ordered elements. For the `array` type, you must also specify `items.type` to specify the data type of the items in the array.
++
+For example:
++
+[source,yaml]
+----
+properties:
+    scopes:
+        type: array
+        items:
+            type: string
+        default: ["READ", "WRITE"]
+----
++
+Object::
++
+The `object` type specifies the data as a list of ordered elements with multiple properties. For the `object` type, you must also specify `properties.type` for each element in the object.
++
+For example:
++
+[source,yaml]
+----
+properties:
+    user:
+      type: object
+      properties:
+        id:
+          type: string
+        name:
+          type: string
+      required:
+        - id
+        - name
+      default: {"id": "1", "name": "John Doe"}
+----
+
+== Use Custom Parameters in Rust Source Code
+
+After you modify the `gcl.yaml` file you can propagate the changes to the `src/generated/config.rs` rust file and the json schema for the UI by running the `build-asset-files` command:
+
+[source,cmd]
+----
+make build-asset-files
+----
+
+Additionally compiling the project with the `build` command also propagates the changes:
+
+[source,cmd]
+----
+make build
+----
+
+For more information about compiling your policy, see xref:policies-pdk-compile-policies.adoc[].
+
+Properties defined in the `gcl.yaml` files are mapped to the rust data types in the `config.rs` rust file as follows:
```

Comment:
```suggestion
Properties defined in the `gcl.yaml` files are mapped to the Rust data types in the `config.rs` Rust file as follows:
```
I'm marking Rust as uppercase assuming this is talking about the programming language. Please confirm if this is correct.